datatypes/src/collections/batch_builder.rs

```diff
@@ -0,0 +1,443 @@
+            time_array: None,
+            geo_array: None,
+            num_features,
+            num_coords,
+            num_lines: None,
+            num_polygons: None,
+            num_rings: None,
+            output: None,
+            output_type,
+        }
+    }
+
+    pub fn points(
+        types: HashMap<String, FeatureDataType>,
+        num_features: usize,
+        num_coords: usize,
+    ) -> Self {
+        Self {
+            types,
+            column_arrays: HashMap::new(),
+            time_array: None,
+            geo_array: None,
+            num_features,
+            num_coords,
+            num_lines: None,
+            num_polygons: None,
+            num_rings: None,
+            output: None,
+            output_type: VectorDataType::MultiPoint,
+        }
+    }
+
+    pub fn lines(
+        types: HashMap<String, FeatureDataType>,
+        num_features: usize,
+        num_lines: usize,
+        num_coords: usize,
+    ) -> Self {
+        Self {
+            types,
+            column_arrays: HashMap::new(),
+            time_array: None,
+            geo_array: None,
+            num_features,
+            num_coords,
+            num_lines: Some(num_lines),
+            num_polygons: None,
+            num_rings: None,
+            output: None,
+            output_type: VectorDataType::MultiLineString,
+        }
+    }
+
+    pub fn num_features(&self) -> usize {
+        self.num_features
+    }
+
+    pub fn num_coords(&self) -> usize {
+        self.num_coords
+    }
+
+    pub fn num_lines(&self) -> Result<usize> {
+        self.num_lines
+            .ok_or_else(|| FeatureCollectionError::WrongDataType.into())
+    }
+
+    pub fn num_polygons(&self) -> Result<usize> {
+        self.num_polygons
+            .ok_or_else(|| FeatureCollectionError::WrongDataType.into())
+    }
+
+    pub fn num_rings(&self) -> Result<usize> {
+        self.num_rings
+            .ok_or_else(|| FeatureCollectionError::WrongDataType.into())
+    }
+
+    pub fn column_types(&self) -> HashMap<String, FeatureDataType> {
+        self.types.clone()
+    }
+
+    pub fn set_time_intervals(&mut self, values_buffer: Buffer) -> Result<()> {
+        let data = ArrayData::builder(TimeInterval::arrow_data_type())
+            .len(self.num_features)
+            .add_buffer(values_buffer)
+            .build();
+
+        let array = Arc::new(FixedSizeListArray::from(data)) as ArrayRef;
+
+        self.time_array = Some(array);
+
+        Ok(())
+    }
+
+    pub fn set_default_time_intervals(&mut self) -> Result<()> {
+        let mut time_intervals_builder = TimeInterval::arrow_builder(self.num_features);
+
+        let default = TimeInterval::default();
+        for _ in 0..self.num_features {
+            let date_builder = time_intervals_builder.values();
+            date_builder.append_value(default.start().inner())?;
+            date_builder.append_value(default.end().inner())?;
```

Comment:
You could extract the two ints once before the loop.

Reply:
72ac38a30e27628e999209f40b1d6d8c20a95b0f